apps/designer/app/designer/shared/design-tokens-manager/design-tokens-manager.tsx

```diff
@@ -79,108 +104,230 @@ const TokenEditor = ({
   }, [isOpen, errors.hasErrors]);
 
   const handleChange = (event: FormEvent<HTMLFormElement>) => {
-    if (errors.hasErrors === false) return;
-    const data = getData(event);
-    const nextErrors = validate(tokens, data);
+    if (errors.hasErrors === false || formRef.current === null) {
+      return;
+    }
+    const data = getData(formRef.current);
+    const nextErrors = validate(tokens, data, isNew);
     setErrors(nextErrors);
   };
 
-  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    const data = getData(event);
-    const nextErrors = validate(tokens, data);
+  const handleSubmit = (event?: FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+    if (formRef.current === null) return;
+    const data = getData(formRef.current);
+    const nextErrors = validate(tokens, data, isNew);
     setErrors(nextErrors);
 
     if (nextErrors.hasErrors === false) {
-      onChangeComplete({ ...data, group, type } as DesignToken);
-      setIsOpen(false);
+      onChangeComplete({ ...token, ...data } as DesignToken);
```

Comment:
Why cast to DesignToken?

Reply:
because data is not necessary a correct token, so not sure how to handle it, either I will have to cast at getData or here, its basically serialization of the form

Reply:
moved casting to where we get the data, so that its obvious why we need t cast, does that make sense now?